### PR TITLE
Added docs for Config component

### PR DESCRIPTION
**Pull request by mickaelandrieu**
This is docs for the BackBee Config Component

REFS https://github.com/backbee/backbee-standard/issues/223 and  https://github.com/backbee/backbee-standard/issues/170

### CODE REVIEW

File: doc/developper-documentation/components/config.md [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+* ``persistor`` is the list of activated persistor, BackBee will loop on each and try to find configuration. If no one 
+  provide the requested configuration, an exception will be throw. 
+
+### Configuration caching
+
+The BackBee Config Component is fully compatible with BackBee Cache component.
+
+When you create your Config object, you can pass an instance of ``BackBee\Cache\AbstractCache`` as a second argument.
+This way, in non-debug mode (fourth argument, **false** by default), the second time you get the information, the configuration
+will be constructed from cache instead of built again from files or database registry.
```
Comment (crouillon) [inline doc/developper-documentation/components/config.md]: :+1: 
I think a chapter should be add to speak about the public method extend($basedir = null, $overwrite = false), it's a key point to allow the overriding of the config

File: doc/developper-documentation/components/config.md [hunk 2]
```diff
@@ -0,0 +1,137 @@
+# Components
+
+## The Config component
+
+This component is used in BackBee to set services and parameters from YAML files into the Container.
```
Comment (crouillon) [inline doc/developper-documentation/components/config.md]: not services, only parameters values for services

Comment (mickaelandrieu) [inline doc/developper-documentation/components/config.md]: we can't declare new services in YAML ?

Comment (crouillon) [inline doc/developper-documentation/components/config.md]: yes we can but the Config component doesn't set them

**Comment by mickaelandrieu** [comment]
Thanks for review @crouillon

**Pull request merged**